bazel/enkit_credential_helper/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/enfabrica/enkit/lib/config/defcon"
+	"github.com/enfabrica/enkit/lib/config/identity"
+)
+
+type Credentials struct {
+	Headers map[string][]string `json:"headers"`
+}
+
+func exitIf(err error) {
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	if len(os.Args) <= 1 {
+		exitIf(fmt.Errorf("no command given"))
+	}
+	switch os.Args[1] {
+	case "get":
+	default:
+		exitIf(fmt.Errorf("bad command %q", os.Args[1]))
+	}
+
+	store, err := identity.NewStore("enkit", defcon.Open)
+	exitIf(err)
+
+	_, token, err := store.Load("")
+	exitIf(err)
+
+	creds := &Credentials{
+		Headers: map[string][]string{
+			"cookie": []string{fmt.Sprintf("Creds=%s", token)},
```

Comment:
can you use `http.Cookie` `.String()` to generate the header? The RFC mandates which characters are valid in a cookie, and how to escape the ones that are not valid. I don't think we can guarantee that the token - as is - is RFC/HTTP compliant. I'm worried the creds helper will work in most cases, but whenever the cookie contains some invalid character, it won't work as it's not escaped correctly.

Reply:
Done! This actually turns out better because it eliminates the `fmt.Sprintf`